src/plugins/discover/public/application/main/discover_main_app.tsx

```diff
@@ -101,10 +98,8 @@ export function DiscoverMainApp(props: DiscoverMainProps) {
   useSavedSearchAliasMatchRedirect({ savedSearch, spaces, history });
 
   return (
-    <DiscoverAppStateProvider value={stateContainer.appStateContainer}>
+    <DiscoverMainProvider value={stateContainer}>
```

Comment:
Here's where `AppState` & `InternalState` providers get the stateContainer assingned